test/common/upstream/health_check_fuzz.proto

```diff
@@ -0,0 +1,40 @@
+syntax = "proto3";
+
+package test.common.upstream;
+
+import "validate/validate.proto";
+import "test/fuzz/common.proto";
+import "envoy/config/core/v3/health_check.proto";
+
+message Respond {
+  test.fuzz.Headers headers = 1;
+  string status = 2 [(validate.rules).string.pattern = "[0-9]+$*"];
```

Comment:
nit: add comment on the pattern and link to http_parser for why.

also, something like abs0000 will give a match. it needs to match at the start of the string, and include a space, since http_parser might accept that. what is the * after the end of word $ matcher? does that compile?